be/src/vec/exec/volap_scan_node.cpp

```diff
@@ -554,8 +575,11 @@ Block* VOlapScanNode::_alloc_block(bool& get_free_block) {
 int VOlapScanNode::_start_scanner_thread_task(RuntimeState* state, int block_per_scanner) {
     std::list<VOlapScanner*> olap_scanners;
     int assigned_thread_num = _running_thread;
-    size_t max_thread = std::min(_volap_scanners.size(),
-                                 static_cast<size_t>(config::doris_scanner_thread_pool_thread_num));
+    size_t max_thread = config::doris_scanner_queue_size;
+    if (config::doris_scanner_row_num > state->batch_size()) {
```

Comment:
This logic need refine. Actually, no one would change this `doris_scanner_row_num` config, and this `if` condition is always true. So that this `if` condition is meaningless.

Reply:
I think we need this if condition check. If we do not allow anyone modify config:: doris_scanner_row_num, I think it should be a const expr variable not in config file. If it is in conf file, then it maybe modified by someone, so that we should check it.
And also, maybe nobody will modify doris_scanner_row_num, but somebody could modify batch size, the batch size maybe larger than the config::doris_scanner_row_num.